.github/workflows/ode-toolbox-build.yml

```diff
@@ -47,31 +47,33 @@ jobs:
 
     steps:
       - name: Checkout ODE-toolbox code
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
       - name: Setup Python
-        uses: actions/setup-python@v4
+        uses: actions/setup-python@v5
         with:
-          python-version: 3.8
+          python-version: 3.9
 
       - name: Install apt dependencies
         run: |
           sudo apt update
           sudo apt install pkg-config python3-all-dev
 
       - name: Install GSL
+        if: ${{ matrix.with_gsl == '1' }}
         run: |
+          sudo apt remove libgsl27 libgsl-dev # remove system-wide GSL version that is installed by default
           cd ..
-          wget http://ftp.wrz.de/pub/gnu/gsl/gsl-2.5.tar.gz
-          tar -xzf gsl-2.5.tar.gz
-          cd gsl-2.5
+          wget https://mirror.ibcp.fr/pub/gnu/gsl/gsl-2.7.tar.gz # this should be "gsl-latest.tar.gz", but there is an issue with GSL 2.8 (see https://github.com/pygsl/pygsl/issues/62)
+          tar -xzf gsl-2.7.tar.gz
+          cd gsl-2.7
           ./configure && make && sudo make install
 
       - name: Install Python dependencies
         run: |
           python -m pip install --upgrade pip pytest pycodestyle codecov pytest-cov wheel
           python -m pip install numpy
-          if [ "${{ matrix.with_gsl }}" == "1" ]; then python -m pip install pygsl ; fi
+          if [ "${{ matrix.with_gsl }}" == "1" ]; then python3 -m pip install -v https://github.com/pygsl/pygsl/archive/refs/tags/v2.4.1.tar.gz ; fi
```

Comment:
Can this be just a normal `pip install`?

Reply:
Yes, however, the latest PyPI release for PyGSL is outdated, hence the temporary command.

For the upstream tracking issue, please see https://github.com/pygsl/pygsl/issues/59

I have added a note referencing this issue in the Actions yaml file.